chef/data_bags/crowbar/migrate/provisioner/020_upgrade_autoyast_repos.rb

```diff
@@ -0,0 +1,21 @@
+def upgrade ta, td, a, d
+  if !a['suse']['autoyast']['repos'].nil? && !a['suse']['autoyast']['repos'].empty?
+    if %w(common suse-11.3 suse-12.0).select {|k|
+        a['suse']['autoyast']['repos'].keys.include? k
+    }.empty?
+      repos = a['suse']['autoyast'].delete('repos')
+      a['suse']['autoyast']['repos'] = {}
+      a['suse']['autoyast']['repos']['suse-11.3'] = repos
+    end
+  end
+  return a, d
+end
+
+def downgrade ta, td, a, d
+  if !a['suse']['autoyast']['repos']['suse-11.3'].nil? && !a['suse']['autoyast']['repos']['suse-11.3'].empty?
+    a['suse']['autoyast']['repos']      = a['suse']['autoyast']['repos']['suse-11.3']
+  elsif !a['suse'].nil?
+    a.delete('suse')
```

Comment:
I don't think we need that `elsif` bit. Or maybe I'm missing the case where we need it?